### ISSUE TITLE
Enable auto provision of TLS certificates with Let's go!

### ISSUE THREAD

**Issue by sesispla** [sesispla/generator-kubegen#2]
The Ingress Controller can be enhanced with an automatic Let's go! TLS certificate using kube-lego.

**Questions**
```bash
(Ingress) (Ingress) Which class of expose would you like?
[ ] external
[ ] internal
[ ] TLS
(Ingress) Which Certificate Authority would you like to use?
[X] Let's Encrypt
(Ingress) Let's Encrypt Account e-mail
you@mail.com
(Ingress) Secret Storage Name
Default: {answers.name}-tls
```

The generated YAML file for the Ingress should look like this:

```yml
# nginx/ingress.yml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
    name: nginx
    namespace: default
    annotations:        
        kubernetes.io/tls-acme: "true"
spec:
  tls:
  - hosts:
    - mydomain.com
    secretName: nginx-tls
  rules:
    -   host: mydomain.com
        http:
            paths:
                -
                    path: /
                    backend:
                        serviceName: nginx
                        servicePort: 80
```

Additionally, a kibe-lego deployment is required:

```yml
# kube-lego/deployment.yml
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: kube-lego
  namespace: kube-lego
spec:
  replicas: 1
  template:
    metadata:
      labels:
        app: kube-lego
    spec:
      containers:
      - name: kube-lego
        image: jetstack/kube-lego:0.1.4
        imagePullPolicy: Always
        ports:
        - containerPort: 8080
        env:
        - name: LEGO_EMAIL
          valueFrom:
            configMapKeyRef:
              name: kube-lego
              key: lego.email
        - name: LEGO_URL
          valueFrom:
            configMapKeyRef:
              name: kube-lego
              key: lego.url
        - name: LEGO_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
        - name: LEGO_POD_IP
          valueFrom:
            fieldRef:
              fieldPath: status.podIP
        readinessProbe:
          httpGet:
            path: /healthz
            port: 8080
          initialDelaySeconds: 5
          timeoutSeconds: 1
```

```yml
# kube-lego/configmap.yml
apiVersion: v1
kind: ConfigMap
metadata:
  name: kube-lego
  namespace: kube-lego
data:
  # modify this to specify your address
  lego.email: "you@mail.com"
  # configure letsencrypt's production api
  lego.url: "https://acme-v01.api.letsencrypt.org/directory"
```

**Comment by sesispla** [comment]
Added a new sub-generator to create all the kube-lego stuff directly from the generator:

```bash
$ yo kubegen:lego

 |  |/  / |  |  |  | |   _  \  |   ____| /  _____||   ____||  \ |  |
 |  '  /  |  |  |  | |  |_)  | |  |__   |  |  __  |  |__   |   \|  |
 |    <   |  |  |  | |   _  <  |   __|  |  | |_ | |   __|  |  . `  |
 |  .  \  |  `--'  | |  |_)  | |  |____ |  |__| | |  |____ |  |\   |
 |__|\__\  \______/  |______/  |_______| \______| |_______||__| \__|

Welcome to Kubernetes Generator (kubegen)!
? How the service should be named? kube-lego
? In which Namespace should be deployed? kube-system
? (Let's Encrypt) Which Docker image should the Deployment use? jetstack/kube-lego:0.1.4
? (Let's Encrypt) Your Let's Encrypt e-mail? sesispla@outlook.com
? (Let's Encrypt) Is the Let's Encrypt service URL correct? https://acme-v01.api.letsencrypt.org/directory
   create deployment.yml
   create configmap.yml
```

Once the kibe-lego is deployed to the cluster, from Ingress' first question you can choose the option "tls-lego", what will configure the Ingress to use kube-lego

```bash
Sergios-MacBook-Pro:generator-kubegen-tests Sergio$ yo kubegen

 |  |/  / |  |  |  | |   _  \  |   ____| /  _____||   ____||  \ |  |
 |  '  /  |  |  |  | |  |_)  | |  |__   |  |  __  |  |__   |   \|  |
 |    <   |  |  |  | |   _  <  |   __|  |  | |_ | |   __|  |  . `  |
 |  .  \  |  `--'  | |  |_)  | |  |____ |  |__| | |  |____ |  |\   |
 |__|\__\  \______/  |______/  |_______| \______| |_______||__| \__|

Welcome to Kubernetes Generator (kubegen)!
? How the service should be named? nginx
? In which Namespace should be deployed? default
? Which type of Pod controller mechanism whould you like to use? Replication Controller
? (Replication Controller) Which Docker image should the Deployment use? nginx
? (Replication Controller) How much container replicas should be created? 1
? (Service) In which port is the Container listening? 80
? (Service) In which port should the Service listen? 80
? (Ingress) Would like to expose the service out of the cluster? yes
? (Ingress) Which class of expose would you like? tls-lego
? (Ingress) Does the service have a hostname? nginx.minikube.sesispla.net
? (Ingress) A name for the kube-lego TLS storage secret? nginx-secret
? (Ingress) Ingress root path? /
? (Ingress) In which port should the Ingress listen? 80
create rc.yml
create svc.yml
create ing.yml
```

**Comment by sesispla** [comment]
Merged changes after feature tests in 66f10c4fd98d0bf166b67132af791082bb9d7c3d
